src/app/components/trip-details/trip-details.component.ts

```diff
@@ -123,23 +123,23 @@ export class TripDetailsComponent {
   }
 
   clearActivityToEdit() {
-    this.activityToEdit.set({} as ItineraryItem);
+    this.activityToEdit.set(null);
   }
 
   updateActivity(activity: ItineraryItem) {
     if (this.tripToUpdate()?.docId && this.tripToUpdate()?.docId !== '') {
-      const newActivities = this.tripToUpdate().itinerary?.map(act =>
+      const newActivities = this.tripToUpdate()?.itinerary?.map(act =>
         act.id === activity.id ? activity : act
       );
       this.store.dispatch(
```

Comment:
I stand corrected but I think it may be better if the newActivities were passed into the action being dispatched. Then the copying of the trip state with the new details will be done in the reducer